app/javascript/gamesFactory.js

```diff
@@ -189,16 +256,27 @@ angular.module('dappChess').factory('games', function (crypto, navigation, gameS
    */
   games.update = function (game) {
     console.log('update game called', game);
-    for(let i in games.list) {
-      if(games.list[i].gameId === game.gameId) {
-        games.list[i] = game;
+    for (let i in games.list) {
+      if (games.list[i].gameId === game.gameId) {
+        // update existing game
+        let g = games.list[i];
+        for (let key in game) {
+          g[key] = game[key];
+        }
```

Comment:
use jQuery.extend(true, g, game) instead if you want a deep copy, or jQuery.extend(g, game) if you want to replace sub-objects